nixos/prometheus-snmp-exporter: add environmentFile option

asbish opened the pull request:
## Description of changes

Prometheus snmp-exporter has support to pass sensitive data as environment variables. Since other exporter configurations for NixOS have `environmentFile` option, the same option is added to snmp-exporter.

As `services.prometheus.exporters.snmp.configurationPath` option can no longer accept paths that are not on the build host in NixOS 24.05, I think this `environmentFile` option would be a substitute for passing sensitive data. 

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

asbish committed:
nixos/prometheus-snmp-exporter: add environmentFile option

Prometheus snmp-exporter has support to pass sensitive data as environment
variables. Since other exporter configurations for NixOS have
environmentFile option, the same option is added to snmp-exporter.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4202

SuperSandro2000 commented on nixos/modules/services/monitoring/prometheus/exporters/snmp.nix:
```diff
@@ -81,6 +81,32 @@ in
+      default = null;
+      example = "/root/prometheus-snmp-exporter.env";
+      description = ''
+        Environment file as defined in {manpage}`systemd.exec(5)`.
```
```suggestion
        EnvironmentFile as defined in {manpage}`systemd.exec(5)`.
```

SuperSandro2000 commented on nixos/modules/services/monitoring/prometheus/exporters/snmp.nix:
```diff
@@ -81,6 +81,32 @@ in
+
+        For variables to use see [Prometheus Configuration](https://github.com/prometheus/snmp_exporter#prometheus-configuration).
+
+        Besides setting the filepath for this option, you also need to add
+        flag `--config.expand-environment-variables` to `extraFlags`.
+
```
We should do that if the environmentFile is defined to make this option even more useful

asbish committed:
nixos/prometheus-snmp-exporter: add a parameter implicitly

Change to adding `--config.expand-environment-variables` parameter to
`ExecStart` if `environmentFile` option is set.

asbish commented:
@SuperSandro2000 
Thank you for reviewing this PR. I've made some changes based on your feedback.
